src/privileged/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,353 @@
+        Year: (new Date(site.BreachDate)).getFullYear(),
+      });
+    });
+
+    // Refresh every hour.
+    let one_hour = 60 * 60 * 1000;
+    this._loadBreachesTimer = setTimeout(() => this.loadBreaches(), one_hour);
+  },
+
+  // nsIWebProgressListener implementation.
+  onStateChange(aBrowser, aWebProgress, aRequest, aStateFlags, aStatus) {
+    let location = aRequest.URI;
+
+    if (!aWebProgress.isTopLevel || aWebProgress.isLoadingDocument ||
+        !Components.isSuccessCode(aStatus)) {
+      return;
+    }
+
+    let host;
+    try {
+      host = Services.eTLD.getBaseDomain(location);
+    } catch (e) {
+      // If we can't get the host for the URL, it's not one we
+      // care about for breach alerts anyway.
+      return;
+    }
+
+    this.warnIfNeeded(aBrowser, host);
+  },
+
+  startObserving() {
+    if (this.observerAdded) {
+      return;
+    }
+
+    EveryWindow.registerCallback(
+      this.kNotificationID,
+      (win) => {
+        // Inject our stylesheet.
+        let DOMWindowUtils = win.windowUtils;
+        if (!DOMWindowUtils) {
+          // win.windowUtils was added in 63, fallback if it's not available.
+          DOMWindowUtils = win.QueryInterface(Ci.nsIInterfaceRequestor)
+                              .getInterface(Ci.nsIDOMWindowUtils);
+        }
+        DOMWindowUtils.loadSheetUsingURIString(this.getURL("privileged/FirefoxMonitor.css"),
+                                               DOMWindowUtils.AUTHOR_SHEET);
+
+        // Set up some helper functions on the window object
+        // for the popup notification to use.
+        win.FirefoxMonitorUtils = {
+          // Keeps track of all notifications currently shown,
+          // so that we can clear them out properly if we get
+          // disabled.
+          notifications: new Set(),
+          disable: () => {
+            this.disable();
+          },
+          getString: (aKey) => {
+            return this.getString(aKey);
+          },
+          getFormattedString: (aKey, args) => {
+            return this.getFormattedString(aKey, args);
+          },
+        };
+
+        // Setup the popup notification stuff. First, the URL bar icon:
+        let doc = win.document;
+        let box = doc.getElementById("notification-popup-box");
+        // We create a box to use as the anchor, and put an icon image
+        // inside it. This way, when we animate the icon, its scale change
+        // does not cause the popup notification to bounce due to the anchor
+        // point moving.
+        let box2 = doc.createElementNS(XUL_NS, "box");
+        box2.setAttribute("id", `${this.kNotificationID}-notification-anchor`);
+        box2.classList.add("notification-anchor-icon");
+        let img = doc.createElementNS(XUL_NS, "image");
+        img.setAttribute("role", "button");
+        img.classList.add(`${this.kNotificationID}-icon`);
+        box2.appendChild(img);
+        box.appendChild(box2);
```

Comment:
`box` and `box2` aren't great names for two things that are not two versions of the exact same thing, please make them more descriptive.